sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -59,6 +59,32 @@ object HiveTypeCoercion {
   }
 
   /**
+   * Implicit promote the AtomicType to StringType if
+   * one of the data type in (dt1, dt2) is StringType, and the other is not either
+   * BooleanType or BinaryType, the TightestCommonType should be StringType
+   * eg: 1. CaseWhenLike  case when ... then dt1 else dt2 end
+   *     2. Coalesce(null, dt1, dt2)
+   */
+  private def promoteToStringType(t1: DataType, t2: DataType): Option[DataType] = (t1, t2) match {
```

Comment:
@cloud-fan I'm sorry, I don't quite understand when to use a function or when to use a method. And what's the purpose here to use a function.